src/openforms/submissions/api/validation.py

```diff
@@ -105,10 +105,13 @@ def get_invalid_prefilled_fields(
 
         plugin_name = component["prefill"]["plugin"]
         attribute_name = component["prefill"]["attribute"]
-        if (
-            submission_step.data[component["key"]]
-            != submission.prefill_data[plugin_name][attribute_name]
-        ):
+
+        if submission.prefill_data[plugin_name][attribute_name] is None:
+            initial_value = component["defaultValue"]
+        else:
+            initial_value = submission.prefill_data[plugin_name][attribute_name]
+
+        if submission_step.data[component["key"]] != initial_value:
```

Comment:
```suggestion

        if submission.prefill_data[plugin_name][attribute_name] is None:
            # the value will be `None` if there is no actual prefill data available, so there is nothing to compare to. This
            # especially applies to test-environments without real prefill-connections.
            continue
```
perhaps? though I do realize this leaves open the option to spoof data